content/productivity/repeat-attack/plugin.js

```diff
@@ -0,0 +1,192 @@
+const { default: Manager } = await import(
+  "https://plugins.zkga.me/utils/RepeatAttackCore.js"
+);
+
+const { html, render, useState, useLayoutEffect } = await import(
+  "https://unpkg.com/htm/preact/standalone.module.js"
+);
+
+let Spacing = {
+  marginLeft: "12px",
+  marginRight: "12px",
+};
+let VerticalSpacing = {
+  marginBottom: "12px",
+};
+let Clickable = {
+  cursor: "pointer",
+  textDecoration: "underline",
+};
+let ActionEntry = {
+  marginBottom: "10px",
+  display: "flex",
+  justifyContent: "space-between",
+  color: "",
+};
+
+function centerPlanet(id) {
+  ui.centerLocationId(id);
+}
+
+function Attack({ action }) {
+  return html`
+    <div key=${action.id} style=${ActionEntry}>
+      <span>
+        <span
+          style=${{ ...Spacing, ...Clickable }}
+          onClick=${() => centerPlanet(action.payload.srcId)}
+          >${action.payload.srcId.substring(5, 10)}</span
+        >
+        →
+        <span
+          style=${{ ...Spacing, ...Clickable }}
+          onClick=${() => centerPlanet(action.payload.syncId)}
+          >${action.payload.syncId.substring(5, 10)}</span
+        ></span
+      >
+      <button
+        onClick=${() => {
+          op.delete(action.id);
+        }}
+      >
+        ✕
+      </button>
+    </div>
+  `;
+}
+
+function AddAttack() {
+  let [planet, setPlanet] = useState(ui.getSelectedPlanet());
+  let [source, setSource] = useState(false);
+  let [target, setTarget] = useState(false);
+  let onClick = () => {
+    setPlanet(ui.getSelectedPlanet());
+  };
+  useLayoutEffect(() => {
+    window.addEventListener("click", onClick);
+    return () => {
+      window.removeEventListener("click", onClick);
+    };
+  }, [onClick]);
+
+  function createAttack(source, target) {
+    op.pester(source.locationId, target.locationId);
+  }
+
+  return html`
+    <div>
+      <button
+        style=${VerticalSpacing}
+        onClick=${() => {
+          setSource(planet);
+        }}
+      >
+        Set Source
+      </button>
+      <span style=${{ ...Spacing }}
+        >${source ? source.locationId.substring(5, 10) : ""}</span
+      >
+    </div>
+    <div>
+      <button
+        style=${VerticalSpacing}
+        onClick=${() => {
+          setTarget(planet);
+        }}
+      >
+        Set Target
+      </button>
+      <span style=${{ ...Spacing }}
+        >${target ? target.locationId.substring(5, 10) : ""}</span
+      >
+    </div>
+    <button
+      style=${VerticalSpacing}
+      onClick=${() => createAttack(source, target)}
+    >
+      submit
+    </button>
+  `;
+}
+
+function AttackList() {
+  const [actions, setActions] = useState(op.actions);
+
+  let actionList = {
+    maxHeight: "100px",
+    overflowX: "hidden",
+    overflowY: "scroll",
+  };
+
+  let actionsChildren = actions
+    .filter((a) => a.type == "PESTER")
+    .map((action) => {
+      return html`<${Attack} action=${action}></${Attack}>`;
```

Comment:
Oooo, I didn't know closing worked like that. This looks like it can be self-closing though?
```suggestion
      return html`<${Attack} action=${action} />`;
```